spicy/toolchain/bin/spicy-doc.cc

```diff
@@ -116,54 +116,74 @@ static json operandToJSON(const hilti::operator_::Operand& o) {
-            else if ( o.kind() == hilti::operator_::Kind::MemberCall ) {
-                auto operands = o.operands();
-                auto self = operands[0];
-                auto method = std::get<hilti::Type>(operands[1].type).as<hilti::type::Member>();
-
-                if ( ! std::get<hilti::Type>(operands[2].type).isA<hilti::type::OperandList>() )
-                    continue; // XXX
-
-                auto params = std::get<hilti::Type>(operands[2].type).as<hilti::type::OperandList>();
-
-                operator_["self"] = operandToJSON(self);
-                operator_["id"] = method.id();
-                operator_["args"] = std::list<json>();
-
-                for ( const auto& p : params.operands() )
-                    operator_["args"].push_back(operandToJSON(p));
-            }
-            else {
-                operator_["operands"] = json();
-
-                for ( const auto& x : o.operands() )
-                    operator_["operands"].push_back(operandToJSON(x));
-            }
-
-            all_operators.push_back(std::move(operator_));
+            auto params = std::get<hilti::Type>(operands[2].type).as<hilti::type::OperandList>();
+
+            jop["self"] = operandToJSON(self);
+            jop["id"] = method.id();
+            jop["args"] = std::list<json>();
+
+            for ( const auto& p : params.operands() )
+                jop["args"].push_back(operandToJSON(p));
         }
-    }
+        else {
+            jop["operands"] = json();
+
+            for ( const auto& x : op.operands() )
+                jop["operands"].push_back(operandToJSON(x));
+        }
+
+        all_operators.push_back(std::move(jop));
+    };
+
+    // Iterate through all available operators.
+    auto operators = hilti::operator_::registry().all();
+    for ( const auto& [kind, operators] : operators )
+        for ( const auto& op : operators )
+            add_operator(op.docNamespace(), op);
+
+    // Hardcode concrete instances of generic operators. They need to be
+    // associated with the corresponding types, but there's no generic way to
+    // do that.
+    add_operator("bytes", hilti::operator_::generic::Begin::Operator());
+    add_operator("bytes", hilti::operator_::generic::End::Operator());
+    add_operator("list", hilti::operator_::generic::Begin::Operator());
+    add_operator("list", hilti::operator_::generic::End::Operator());
+    add_operator("map", hilti::operator_::generic::Begin::Operator());
+    add_operator("map", hilti::operator_::generic::End::Operator());
+    add_operator("set", hilti::operator_::generic::Begin::Operator());
+    add_operator("set", hilti::operator_::generic::End::Operator());
+    add_operator("stream", hilti::operator_::generic::Begin::Operator());
+    add_operator("stream", hilti::operator_::generic::End::Operator());
+    add_operator("vector", hilti::operator_::generic::Begin::Operator());
+    add_operator("vector", hilti::operator_::generic::End::Operator());
```

Comment:
We could use a loop here to make this slightly more maintainable,

```suggestion
    for ( auto& type_ : std::vector({"bytes", "list", "map", "set", "stream", "vector"}) ) {
        add_operator(type_, hilti::operator_::generic::Begin::Operator());
        add_operator(type_, hilti::operator_::generic::End::Operator());
    }
```